terminal/noninteractive.go

```diff
@@ -212,6 +215,51 @@ func (ui *nonInteractiveUI) Error(msg string) {
 func (ui *nonInteractiveUI) ErrorWithContext(err error, sub string, ctx ...string) {
 	ui.Error(helper.Title(sub))
 	ui.Error("  Error: " + err.Error())
+
+	// Selectively promote Details and Suggestion from the context.
+	var extractItem = func(ctx []string, key string) ([]string, string, bool) {
+		for i, v := range ctx {
+			if strings.HasPrefix(v, key) {
+				outStr := v
+				outCtx := slices.Delete(ctx, i, i+1)
+				return outCtx, outStr, true
+			}
+		}
+		return ctx, "", false
+	}
```

Comment:
It seems like `extractItem` is shared in 3 places but the `testui` makes that hard to extract usefully. I wouldn't bother in this PR, but sometime in the future we might want to think about whether there's a set of common "ui helpers" we can factor out of these methods so we're not having to repeat them. (Ex. can noninteractive and testui share most of their code if they both take a writer that's configured by the caller.)

Definitely future work though!

Reply:
Just noticed that I didn't update the glint UI implementation of `promote` to look like the non-interactive one. (It came later and that was when I heard @pkazmierczak's voice in my head, "Use strings.Cut()...".  Since it doesn't break anything, will make those the same in a follow-up PR.  Especially since I still want to come up with a hanging indent and wrapping story for the glint ui later anyway.